code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -404,6 +404,8 @@ This saves us from having to call add_fingerprint() any time something is put in
 /mob/living/carbon/human/put_in_hands(var/obj/item/W)
 	if(!W)
 		return 0
+	if(W in src.contents)
+		u_equip(W)
```

Comment:
Check out the comment above this proc definition, apparently it shouldn't be called directly.

Also checking if W in contents isn't necessary.